gui/default/syncthing/core/syncthingController.js

```diff
@@ -240,6 +240,13 @@ angular.module('syncthing.core')
             }
         });
 
+        $scope.$on(Events.FOLDER_WATCHER_STATE_CHANGED, function (event, arg) {
```

Comment:
What if the event rolled off? How often is this re-emit? Why not have this part of some folder completion status call or some shit like that?

Reply:
Perhaps part of folder summary service?

Reply:
It already is part of that and updated when incoming as well. Unfortunately a fs watcher error does not trigger a new folder summary and the only way it can trigger is by emitting an event. And if there is an event anyway, I thought I might as well use it in the UI. This trigger could however be removed without big drawbacks (folder summaries are probably a bit slower generally and might even get delayed by a minute (?) worst case).

Reply:
Syncthing stores last 1k or so events. If you have a lot of tiny files being downloaded, FileStarted, FileFinished will force your event to roll off and it will never be visible.

Why can't your watcher emit an event, but that even be picked up by the summary service and the summary service sets the error on the summary, and then you check if the error is in the summary.

Reply:
That happens too, `scope.model` is the folder summary and contains the fs watcher error. This event handling is just to speed things up in case the folder summary is delayed. I'll just remove it.

Reply:
It's never delayed by more than 2 or so seconds, Adding an extra event because of that seems like overkill?

Reply:
It's removed since 4 days ;)